components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -623,8 +632,13 @@ private OAuthConsumerAppDTO createOAuthApp(ApplicationRegistrationRequest regist
         oAuthConsumerApp.setRequestObjectSignatureValidationEnabled(registrationRequest.isRequireSignedRequestObject());
         oAuthConsumerApp.setRequirePushedAuthorizationRequests(
                 registrationRequest.isRequirePushedAuthorizationRequests());
-        oAuthConsumerApp.setTlsClientCertificateBoundAccessTokens(
-                registrationRequest.isTlsClientCertificateBoundAccessTokens());
+        if (registrationRequest.isTlsClientCertificateBoundAccessTokens()) {
+            if (Boolean.parseBoolean(IdentityUtil.getProperty(
+                    OAuthConstants.ENABLE_TLS_CERT_BOUND_ACCESS_TOKENS_VIA_BINDING_TYPE))) {
```

Comment:
instead of checking for the config here lets check if the CERTIFICATE_BASED_TOKEN_BINDER component is available and then set it.

Reply:
changed